### PR TITLE
Add support to show close button for active tab only.

### LINKED ISSUE #13672: Add `tab.closeButton: activeOnly`
`TabView` is a part of WinUI, and that code is available over at https://github.com/microsoft/microsoft-ui-xaml/

> I want to know if it's even possible on the current version of `TabView` to actualize what that enum would represent?

Hm. 
https://github.com/microsoft/terminal/blob/a9c3b7724c05a0da059c4a0433ea446f8e19cbff/src/cascadia/TerminalApp/TerminalPage.cpp#L2740-L2772

> **Note**
> ## Walkthrough

* We could take that, turn that into a method. 
* Call that method every time the active tab changes... 
* ... or the set of tabs changes. 
* Get rid of all the calls to `_updateTabCloseButton`, cause that only updates a single tab. 
* Add logic in that method
  - Inside the loop, if the mode is `ActiveOnly`, and the tab is active, then `IsClosable(true)`. else `false`
  - Outside the loop, set `CloseButtonOverlayMode` to `Auto`, probably a sensible default.

That sounds like it would work on paper. Wanna file a PR? 😄

_Originally posted by @zadjii-msft in https://github.com/microsoft/terminal/issues/3335#issuecomment-1203867172_

### PR DESCRIPTION
**Pull request by kovdu**
This MR introduces `activeOnly ` for the `showCloseButton` theme option causing the close button only to appear on the active tab.

This is more or less following the approach explained here https://github.com/orgs/microsoft/projects/686/views/2?pane=issue&itemId=19775774 which indeed just works 😄 .

You notice when switching theme the close buttons is back on all tabs again as well.

Closes #13672

I didn't check specific unit tests for this. I hope by making this MR the pipeline will show if I broke something. Or just let me know if you want me to add something specific for this.

### CODE REVIEW

**Review by zadjii-msft — APPROVED** [review]
Yep that's literally how I'd do it 😝 Thanks!

File: src/cascadia/TerminalApp/TerminalPage.cpp [hunk 1]
```diff
@@ -3109,47 +3109,60 @@ namespace winrt::TerminalApp::implementation
         // Begin Theme handling
         _updateThemeColors();
 
+        const auto tab{ _GetFocusedTab() };
+        _updateTabsCloseButton(tab);
```
Comment (carlos-zamora) [inline src/cascadia/TerminalApp/TerminalPage.cpp:3113]: nit: no extra var necessary
```suggestion
        _updateTabsCloseButton(_GetFocusedTab());
```

**Review by carlos-zamora — APPROVED** [review]
Love it! Thanks for the contribution! Please be sure to...
- [ ] update the [schema](https://github.com/microsoft/terminal/blob/main/doc/cascadia/profiles.schema.json)
- [ ] update the [docs](https://github.com/MicrosoftDocs/terminal)

Verified that this works if you change the `showCloseButton` setting via hot reload.

**Comment by carlos-zamora** [comment]
## Demo

Small demo of the result:
active_only


NOTE: removed the demo from the PR body so that the commit history is a bit cleaner

**Comment by carlos-zamora** [comment]
@kovdu I'm not going to merge this yet to give you some time to apply/respond to my comments, but the PR is good to go! Again, thanks for the contribution!

**Comment by kovdu** [comment]
Thanks for the feedback @carlos-zamora .

Fixed your remarks, updated `profiles.schema.json` to reflect the new option and created an MR for updating the documentation as well: https://github.com/MicrosoftDocs/terminal/pull/660. Not sure what the process is to get that one merged though.

Also verified the new value is correctly applied when hot reloading the settings file.

**Pull request merged**